src/Invictus.Testing/LogicAppException.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Runtime.Serialization;
+using System.Security.Permissions;
+using GuardNet;
+
+namespace Invictus.Testing
+{
+    /// <summary>
+    /// Thrown when a problem occurs during interaction with a logic app running in Azure.
+    /// </summary>
+    [Serializable]
+    public class LogicAppException : Exception
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LogicAppException"/> class.
+        /// </summary>
+        public LogicAppException()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LogicAppException"/> class.
+        /// </summary>
+        /// <param name="message">The message that describes the exception.</param>
+        public LogicAppException(string message) : base(message)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LogicAppException"/> class.
+        /// </summary>
+        /// <param name="message">The message that describes the exception.</param>
+        /// <param name="innerException">The exception that is the cause of the current exception</param>
+        public LogicAppException(string message, Exception innerException) : base(message, innerException)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LogicAppException"/> class.
+        /// </summary>
+        /// <param name="message">The message that describes the exception.</param>
+        /// <param name="logicAppName">The name of the logic app resource running in Azure.</param>
+        /// <param name="resourceGroup">The resource group where the logic app is located.</param>
+        /// <param name="subscriptionId">The ID that identifies the subscription on Azure.</param>
+        public LogicAppException(
```

Comment:
We should always order from least to most specific meaning - subscription, resource group, logic app name, message